portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,28 +32,33 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<NewComment> comments = new ArrayList<>();
-
-  private static class NewComment {
+  private static class Comment {
+    long id;
+    long timestamp;
     String username;
     String comment;
 
-    public NewComment(String username, String comment) {
+    public Comment(long id, long timestamp, String username, String comment) {
+      this.id = id;
+      this.timestamp = timestamp;
       this.username = username;
       this.comment = comment;
     }
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String username = getStringParameter(request, "username");
+    String comment = getStringParameter(request, "comment");
```

Comment:
what if both these are empty? do you still want to write to the datastore?

Reply:
Good point, I stopped it from writing when comment is empty and made an empty name = anonymous.